x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportPutShutdownNodeAction.java

```diff
@@ -135,6 +135,18 @@ public void onFailure(Exception e) {
         });
     }
 
+    private boolean isTypeChangeAllowed(SingleNodeShutdownMetadata existingRecord, PutShutdownNodeAction.Request request) {
```

Comment:
We could probably make this static and unit test it, what do you think?

Reply:
Good call, I switched this to take 2 `Type`s (it doesn't need the whole metadata or request) and added some unit tests.